Source/Charts/Renderers/HorizontalBarChartRenderer.swift

```diff
@@ -332,7 +332,7 @@ open class HorizontalBarChartRenderer: BarChartRenderer
             {
                 guard let dataSet = dataSets[dataSetIndex] as? IBarChartDataSet else { continue }
                 
-                if !shouldDrawValues(forDataSet: dataSet) || !(dataSet.isDrawIconsEnabled && dataSet.isVisible)
+                if !shouldDrawValues(forDataSet: dataSet)
```

Comment:
This should be a guard statement:
```
guard shouldDrawValues(for: dataSet),
    dataSet.isDrawIconsEnabled,
    dataSet.isVisible else 
{
    continue
}
```

Reply:
I have updated all related code to guard statement, have a look please.